homework/src/app.js

```diff
@@ -2,10 +2,192 @@
+favoriteAnimals.splice(1, 0, "Samoyed dog");
+
+// Explain in a comment what do you think the new array will look like.
+    //The array will have 5 elements and the new-added element will be set on index 1
+
+// Log your new array.
+console.log(favoriteAnimals);
+
+// Delete 'giraffe' from the array.
+favoriteAnimals.splice(3,1);
+
   /* OPERATORS */
 
+// 1.
+0 == '0'; // true
+0 == [];  // true
+"0" == [];  // false
+
+// Mathematically, if a == b and b == c, then a == c. Can you explain ...?
+     0 == '0' // it is true because loose equality compares the values without looking types.
+     0 == []  // it is true becuase if we convert the empty array to number we get zero : Number([]) = 0
+     "0" == [] // it is false because there is a truthy and falsy conflict at this equality. 
+
+// 2. 
+const x = [1, 2, 3];
+const y = [1, 2, 3];
+const z = y;
+
+    // x == y,   it is false because x and y reference to different values. 
+    // x === y,  it is false because x and y reference to different values.
+    // z === y,  this is true since they reference to same value.
+    // z == x ,  it is false because x and y reference to different values.
+
+// 3. 
+const o1 = { foo: "bar" };
+const o2 = { foo: "bar" };
+const o3 = o2;
+
+// Show that changing o2 changes o3 (or not) and changing o1 changes o3(or not).
+    // Changing o2 changes o3 since o3 is referenced to value of o2.
+    // But changing o1 doesn`t change o3 since they are not correlated.
+
+//Does the order that you assign (o3 = o2 or o2 = o3) matter?
+    // It does matter since o2 is a constant and cannot be reassigned.
   /* QUESTIONS FROM HYF REPOS */
+
+// 8.1 First declare at least four variables and assign them different data types.
+
+let x = '123';
+
+let y = 123;
+
+let arr = ['John'];
+
+let boo = true;
+
+// 8.2 For each variable write a console.log statement that logs the value
+
+console.log("The value of my variable x is: +x");
+
+console.log("The value of my variable y is: +y");
+
+console.log("The value of my variable arr is: +arr");
+
+console.log("The value of my variable boo is: +boo");
+
+
+
+// 8.3 Now write a console.log statement wherein you first explain in words what you think the type of your variables is.
+
+console.log("The type of my variable x is: string");
+
+console.log("The type of my variable y is: number");
+
+console.log("The type of my variable arr is: object");
+
+console.log("The type of my variable boo is: boolean");
+
+// 8.4 Now use typeof to log the actual type of your variables.
+
+console.log(typeof x);
+
+console.log(typeof y);
+
+console.log(typeof arr);
+
+console.log(typeof boo);
+
+// 8.5 Now compare the types of your different variables with one another.
+
+console.log(typeof x === typeof y); 
+
+console.log(typeof x === typeof arr);
+
+console.log(typeof x === typeof boo);
+
+console.log(typeof y === typeof arr);
+
+console.log(typeof y === typeof boo);
+
+console.log(typeof arr === typeof boo);
+
+
```

Comment:
Correct. Here == (equality operator) and === (identity operator) should make no difference since the output of typeof is always of type string.